UnityProject/Assets/Scripts/Systems/Chemistry/ReagentMix.cs

```diff
@@ -295,6 +295,11 @@ public void Add(ReagentMix b)
 
 		public void Add(Reagent reagent, float amount)
 		{
+			if (amount == 0f)
```

Comment:
Use `Matf..Approximately()` when dealing with floats.

Reply:
is exactly 0 value Getting past in,
is normal covers tiny numbers

Reply:
I've said this before on discord. Fool proof your code.
You don't know what anyone will do in the future so do an approximation because floats cannot use == reliably

Reply:
the is no value covers  it anyway, it will just log error instead